ObjectCreation/AeroFoil.FCMacro

```diff
@@ -0,0 +1,1747 @@
+									###
+def AeroFoil_generate(obj):						###
+    n_ = (obj.multi * (obj.multiParam - 1)) + 1				###
+    AeroFoil_object = AeroFoil(obj)					###
+    if AeroFoil_object.create():					###
+        AeroFoil_object.characterize()					###
+    else:								###
+        return False							###
+    AeroFoil_object.copy(n_)						###
+    return True								###
+									###
+###---------------------------------------------------------------------###
+### 		AEROFOIL MACRO CALLS - Top (check Bottom)		###
+###---------------------------------------------------------------------###
+###########################################################################
+
+
+
+# Constant Variables
+# ------------------------------------------------------------------------------------------------
+
+MAX_REFINE_PARAM = 2
+MAX_QUANTITY_PARAM = 5
+MIN_DATA_POINTS = 10
+NACA_NUMBER_OF_POINTS = 50
+NACA_5_2ND_3RD_DIGITS = ["10", "20", "30", "40", "50", "21", "31", "41", "51"]
+FUNCTIONSARRAY = [
+    "+",
+    "-",
+    "*",
+    "/",
+    "^",
+    "e",
+    "pi",
+    "ln(",
+    "log(",
+    "sqrt(",
+    "sin(",
+    "cos(",
+    "tan(",
+    "asin(",
+    "acos(",
+    "atan(",
+    "x",
+    "(",
+    ")",
+    ".",
+]
+UNITSCONVERSION = [0, 1, 10, 1000, 25.4, 304.8, 914.4]
+MIN_CHORD_LENGTH_MM = 1
+MD_ = App.getUserMacroDir(True) + "/AeroFoil_UI_Files/AeroFoil_"
+
+
+
+# Global Variables
+# ------------------------------------------------------------------------------------------------
+
+# Variables Pertaining to the Entire Program
+dialogIndex = [1, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
+
+
+
+# Main Function Class
+# ------------------------------------------------------------------------------------------------
+
+class AeroFoil:
+    def __init__(self, obj):
+        global UNITSCONVERSION
+        self.profileName = generateName("AeroFoil")
+        self.chord = obj.chordLength * UNITSCONVERSION[obj.chordUnits]
+        self.designtype = [obj.workbench, obj.curveType]
+        self.designclosed = obj.closed_
+        self.points = []
+        self.points.append(obj.pointsX)
+        self.points.append(obj.pointsY)
+        self.n_units = len(self.points[0])
+        self.objRefData = [
+            obj.airfoilType,
+            obj.airfoil4DNumber,
+            obj.airfoil5DNumber,
+            obj.airfoilProfileType,
+            obj.importFrom,
+        ]
+        self.designprogressbar = obj.progressBar_
+
+    def copy(self, quantity_):
+        if quantity_ != 1:
+            refNum = int(re.findall(r"\d+", self.profileName)[0])
+            FreeCADGui.Selection.clearSelection()
+            FreeCADGui.Selection.addSelection(
+                App.ActiveDocument.getObjectsByLabel(self.profileName)[0]
+            )
+            FreeCADGui.runCommand("Std_Copy")
+            for i in range(quantity_ - 1):
+                FreeCADGui.runCommand("Std_Paste")
+                App.ActiveDocument.getObjectsByLabel(
```

Comment:
You use `getObjectsByLabel`. Does it mean that the user cannot rename the objects in FreeCAD?